p2p_wallet/Scenes/Main/Swap/Providers/SwapWalletsRepository.swift

```diff
@@ -0,0 +1,75 @@
+import Jupiter
+import Combine
+import SolanaSwift
+import Resolver
+
+public enum SwapWalletsState {
+    case initialized
+    case loading
+    case loaded
+    case error
+}
+
+struct SwapWalletsData {
+    let tokens: [SwapToken]
+    let userWallets: [Wallet]
+}
+
+protocol SwapWalletsRepository {
+    var state: AnyPublisher<SwapWalletsState, Never> { get }
+    var tokens: AnyPublisher<SwapWalletsData, Never> { get }
+
+    func load() async throws
+}
+
+final class SwapWalletsRepositoryImpl: SwapWalletsRepository {
+
+    @MainActor var state: AnyPublisher<SwapWalletsState, Never> {
+        stateSubject.receive(on: RunLoop.main).eraseToAnyPublisher()
+    }
+
+    var tokens: AnyPublisher<SwapWalletsData, Never> {
+        dataSubject.eraseToAnyPublisher()
+    }
+
+    // MARK: - Dependencies
+    private let jupiterClient: JupiterAPI
+    private let localProvider: SwapWalletsProvider
+    @Injected private var walletsRepository: WalletsRepository
+
+    // MARK: - Private params
+    @Published private var stateSubject = CurrentValueSubject<SwapWalletsState, Never>(.loading)
```

Comment:
`@Published` is a short-hand property wrappers for `CurrentValueSubject<ResultType, Never>`. I don't really get it why `stateSubject` and `dataSubject` is both `@Published` of type `CurrentValueSubject` because `@Published` has no effect in this context, the Subject itself never changes, only value changes.
May be you mean `@Published private var state = SwapWalletsState.loading` ?
Or `private var stateSubject = CurrentValueSubject<SwapWalletsState, Never>(.loading)`
They are equivalent

Reply:
I looked through your repository solution and will integrate it as soon as it will be merged into develop

Reply:
I updated PR without your changes yet. Pls look and I will merge it after